src/page-modules/assistant/layout.tsx

```diff
@@ -0,0 +1,222 @@
+  initialToFeature?: GeocoderFeature;
+  initialTransportModesFilter?: TransportModeFilterOption[] | null;
+}>;
+
+function AssistantLayout({
+  children,
+  initialFromFeature,
+  initialToFeature,
+  initialTransportModesFilter,
+}: AssistantLayoutProps) {
+  const { t } = useTranslation();
+  const router = useRouter();
+
+  const [showAlternatives, setShowAlternatives] = useState(false);
+  const [selectedFromFeature, setSelectedFromFeature] = useState<
+    GeocoderFeature | undefined
+  >(initialFromFeature);
+  const [selectedToFeature, setSelectedToFeature] = useState<
+    GeocoderFeature | undefined
+  >(initialToFeature);
+  const [departureDate, setDepartureDate] = useState<DepartureDate>({
+    type: DepartureDateState.Now,
+  });
+  const [transportModeFilter, setTransportModeFilter] = useState(
+    getInitialTransportModeFilter(initialTransportModesFilter),
+  );
+
+  const onSwap = () => {
+    let query = {};
+    const filter = filterToQueryString(transportModeFilter);
+
+    if (filter) {
+      query = {
+        filter,
+      };
+    }
+
+    const arriveBy = departureDate.type === DepartureDateState.Arrival;
+    if (arriveBy) {
+      query = {
+        ...query,
+        arriveBy,
+      };
+    }
+
+    const when = departureDate.type !== DepartureDateState.Now;
+    if (when) {
+      query = {
+        ...query,
+        when: departureDate.dateTime,
+      };
+    }
+    const fromToQuery = featuresToFromToQuery(
+      selectedToFeature,
+      selectedFromFeature,
+    );
+
+    router.push({
+      pathname: `/assistant`,
+      query: {
+        ...query,
+        ...fromToQuery,
+      },
+    });
+  };
+
+  const onSubmitHandler: FormEventHandler<HTMLFormElement> = (e) => {
+    e.preventDefault();
+
+    let query = {};
+
+    const filter = filterToQueryString(transportModeFilter);
+
+    if (filter) {
+      query = {
+        filter,
+      };
+    }
+
+    const arriveBy = departureDate.type === DepartureDateState.Arrival;
+    if (arriveBy) {
+      query = {
+        ...query,
+        arriveBy,
+      };
+    }
+
+    const when = departureDate.type !== DepartureDateState.Now;
+    if (when) {
+      query = {
+        ...query,
+        when: departureDate.dateTime,
+      };
+    }
+
+    const fromToQuery = featuresToFromToQuery(
+      selectedFromFeature,
+      selectedToFeature,
+    );
+
+    router.push({
+      pathname: `/assistant`,
+      query: {
+        ...query,
+        ...fromToQuery,
+      },
+    });
+  };
+
+  return (
+    <div>
+      <form className={style.container} onSubmit={onSubmitHandler}>
+        <div className={style.main}>
+          <div className={style.input}>
+            <Typo.p textType="body__primary--bold" className={style.heading}>
+              {t(PageText.Assistant.search.input.label)}
+            </Typo.p>
+
+            <Search
+              label={t(PageText.Assistant.search.input.from)}
+              onChange={setSelectedFromFeature}
```

Comment:
Should this instead change the URL? Or doesn't that make sense in this case?

Reply:
Is it necessary to change the URL before the "Find departures" button is clicked (or form submitted)?

Reply:
No, that's a good point!